### DISCUSSION TITLE
Is there a way to import a component as a server component without relying on auto-import?

### DISCUSSION THREAD

**Discussion by Mini-ghost** [nuxt/nuxt#30420] [Questions]
Hi there, 

Is there a way to explicitly import a component as a server component without relying on auto-import?

For example, I have a component named `Logo.server.vue`, and it works as a server component when used like this:

```vue
<script setup lang="ts">
// No explicit import of the component
</script>

<template>
  <!-- This will render as a server component -->
  <Logo />
</template>
```

However, if I explicitly import the component, it no longer behaves as a server component:

```vue
<script setup lang="ts">
import Logo from './Logo.server.vue';
</script>

<template>
  <!-- This will NOT render as a server component -->
  <Logo />
</template>
```

Is it possible to enable server component functionality while explicitly importing the component? 

Thank you in advance for your help!

**Comment by Dobefu** [comment]
Hi @Mini-ghost!

Server components in Nuxt use the `NuxtIsland` component under the hood. When importing the component, you can use this component to restore the server component behaviour: https://nuxt.com/docs/api/components/nuxt-island
